apps/dropbox/src/inngest/functions/data-protection/create-sync-shared-links.ts

```diff
@@ -0,0 +1,115 @@
+import { FunctionHandler, inngest } from '@/inngest/client';
+import { getOrganisationAccessDetails } from '../common/data';
+import { InputArgWithTrigger } from '@/inngest/types';
+import { DBXUsers } from '@/connectors';
+import { decrypt } from '@/common/crypto';
+import { ExtendedTeamMemberProfile, SyncJob } from '@/connectors/types';
+import { NonRetriableError } from 'inngest';
+
+const handler: FunctionHandler = async ({
+  event,
+  step,
+}: InputArgWithTrigger<'dropbox/data_protection.shared_link.create.sync_page.requested'>) => {
+  const { organisationId, isFirstSync, syncStartedAt, cursor } = event.data;
+
+  const [organisation] = await getOrganisationAccessDetails(organisationId);
+
+  if (!organisation) {
+    throw new NonRetriableError(
+      `Access token not found for organisation with ID: ${organisationId}`
+    );
+  }
+
+  const token = await decrypt(organisation.accessToken);
+
+  const dbxUsers = new DBXUsers({
+    accessToken: token,
+  });
+
+  const team = await step.run('run-fetch-users', async () => {
+    return dbxUsers.fetchUsers(cursor);
+  });
+
+  if (!team) {
+    throw new Error(`Team is undefined for the organisation ${organisationId}`);
+  }
+
+  const job: SyncJob = {
+    organisationId,
+    syncStartedAt,
+    isFirstSync,
+  };
+
+  const sharedLinkJobs = team.unFormattedMembers.flatMap(({ profile }) => {
+    return [
+      {
+        ...job,
+        teamMemberId: profile.team_member_id,
+        isPersonal: false,
+        pathRoot: (profile as ExtendedTeamMemberProfile).root_folder_id,
+      },
+      {
+        ...job,
+        teamMemberId: profile.team_member_id,
+        isPersonal: true,
+      },
+    ];
+  });
+
+  if (team.members.length > 0) {
+    const eventsToWait = sharedLinkJobs.map(
+      async (sharedLinkJob) =>
+        await step.waitForEvent(`dropbox-wait-sync-shared-links`, {
```

Comment:
```suggestion
        await step.waitForEvent(`wait-sync-shared-links`, {
```

I don't think we need to suffix function step as on inngest we can only see them in a function run.

Reply:
I would like to keep it since I used this pattern in the all Dbx integration

Reply:
A search and replace can changes this for every occurence in a minute.
I think keeping integration consistent is a must have.